### PR TITLE
pattern terminal add auto fill blank pattern

### PR DESCRIPTION
**Pull request by asdflj**
pattern terminal add auto fill blank pattern

add hotkey and baubles support

add avaritia crafting support

### CODE REVIEW

**Comment by asdflj** [comment]
link https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12566

**Comment by LewisSaber** [comment]
Will this also resolve this? https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10614

File: src/main/java/com/glodblock/github/client/gui/container/base/FCContainerEncodeTerminal.java [hunk 1]
```diff
@@ -315,9 +320,40 @@ public void encodeAllItemAndMoveToInventory() {
...
+            int fillStackSize = this.patternSlotIN.getHasStack() ? 64 - this.patternSlotIN.getStack().stackSize : 64;
+            if (fillStackSize == 0) return;
+            for (ItemStack blankPattern : definitions.materials().blankPattern().maybeStack(fillStackSize).asSet()) {
+                IAEItemStack iBlankPattern = AEApi.instance().storage().createItemStack(blankPattern);
+                if (this.patternSlotIN.getHasStack() && !iBlankPattern.isSameType(this.patternSlotIN.getStack()))
+                    continue;
+                IAEItemStack out = this.getHost().getItemInventory()
+                        .extractItems(iBlankPattern, Actionable.SIMULATE, this.getActionSource());
+                if (out != null) {
+                    this.getHost().getItemInventory()
```
Comment (repo-alt) [inline src/main/java/com/glodblock/github/client/gui/container/base/FCContainerEncodeTerminal.java:339]: Perhaps better reassign `out` here, theoretically storage might have been asynchronously updated.

Comment (asdflj) [reply src/main/java/com/glodblock/github/client/gui/container/base/FCContainerEncodeTerminal.java]: fixed

**Review by repo-alt — APPROVED** [review]

**Pull request merged**